src/Ouroboros/Network/Block.hs

```diff
@@ -57,7 +57,7 @@ deriving instance HasHeader block => Eq       (Hash block)
 deriving instance HasHeader block => Ord      (Hash block)
 deriving instance HasHeader block => Show     (Hash block)
 
-instance HasHeader block => Hashable (Hash block)
+instance (HasHeader block, Hashable (HeaderHash block)) => Hashable (Hash block)
```

Comment:
I am right that the `UndecidableInstance` is only because ghc does not know to what `HeaderHash block` will expand to?

Reply:
No, it's because the size of the constraint is not shrinking in this instance, and so it can't determine that instance resolution will terminate. But that is _very_ often the case.

Reply:
@edsko if you could just document the reason: why the `Hashable` constrain is here. So the future generations will not be puzzled by the `UndecidableInstances`.

Reply:
Sorry, only see this now. Will add that to my next PR.

Reply:
I was late with it too :),  or just  push it directly to master

Reply:
Done.